rln/src/public.rs

```diff
@@ -128,12 +127,16 @@ impl RLN<'_> {
 
         // We set the leaves
         for (i, leaf) in leaves.iter().enumerate() {
-            self.tree.set(i, *leaf)?;
+            self.tree.set(i + index, *leaf)?;
```

Comment:
I assume the underlying tree.set function deals with badly set index (didn't inspect)

Reply:
It does throw an error - https://github.com/vacp2p/zerokit/blob/e0c75ee1816c36ac82de9a7bf2f5c05ba64897f0/utils/src/merkle_tree/merkle_tree.rs#L118-L123

Reply:
We *would* allow users of this func to overwrite certain leaves of the merkle tree if they wanted to, should we probably add a warning somewhere?